platform/search/src/services/BareBonesSearchWorker.js

```diff
@@ -29,10 +29,11 @@
     // {id: domainObject's ID, name: domainObject's name}
     var indexedItems = [];
 
-    function indexItem(id, model) {
+    function indexItem(id, model, type) {
         indexedItems.push({
             id: id,
-            name: model.name.toLowerCase()
+            name: model.name.toLowerCase(),
+            type: type
```

Comment:
Does model not have the type property?

Reply:
Yep! Made the same change for GenericSearchWorker as well.